tests/OTAnalytics/plugin_ui/test_svz_metadata_form.py

```diff
@@ -0,0 +1,256 @@
+    viewmodel = MagicMock(spec=ViewModel)
+
+    # Create a separate mock for the update_svz_metadata method
+    update_mock = Mock()
+
+    # Assign the mock to the viewmodel
+    viewmodel.update_svz_metadata = update_mock
+
+    # Set up other return values
+    viewmodel.get_directions_of_stationing.return_value = MockDirections()
+    viewmodel.get_counting_day_types.return_value = MockCountingDayTypes()
+    viewmodel.get_weather_types.return_value = MockWeatherTypes()
+
+    return viewmodel
+
+
+@pytest.fixture
+def resource_manager() -> Mock:
+    resource_manager = Mock(spec=ResourceManager)
+    resource_manager.get.side_effect = lambda key: {
+        SvzMetadataKeys.LABEL_TK_NUMBER: TK_NUMBER_TEST_NAME,
+        SvzMetadataKeys.LABEL_COUNTING_LOCATION_NUMBER: COUNTING_LOCATION_NUMBER_TEST_NAME,  # noqa
+        SvzMetadataKeys.LABEL_DIRECTION_DESCRIPTION: DIRECTION_DESCRIPTION_TEST_NAME,
+        SvzMetadataKeys.LABEL_HAS_BICYCLE_LANE: HAS_BICYCLE_LANE_TEST_NAME,
+        SvzMetadataKeys.LABEL_IS_BICYCLE_COUNTING: IS_BICYCLE_COUNTING_TEST_NAME,
+        SvzMetadataKeys.LABEL_REMARK: REMARK_TEST_NAME,
+        SvzMetadataKeys.LABEL_X_COORDINATE: COORDINATE_X_TEST_NAME,
+        SvzMetadataKeys.LABEL_Y_COORDINATE: COORDINATE_Y_TEST_NAME,
+    }.get(key, f"Resource for {key}")
+    return resource_manager
+
+
+@pytest.fixture
+def svz_metadata_form(
+    viewmodel: ViewModel, resource_manager: ResourceManager
+) -> SVZMetadataForm:
+    return SVZMetadataForm(viewmodel, resource_manager)
+
+
+class TestSVZMetadataFormUI:
+
+    @pytest.mark.asyncio
+    async def test_form_build_up(
+        self, user: User, svz_metadata_form: SVZMetadataForm
+    ) -> None:
+        """Test that the form builds correctly and displays all fields."""
+
+        @ui.page(ENDPOINT_NAME)
+        def page() -> None:
+            svz_metadata_form.build()
+
+        await user.open(ENDPOINT_NAME)
+
+        # Check that all form elements are visible
+        await user.should_see(TK_NUMBER_TEST_NAME)
+        await user.should_see(COUNTING_LOCATION_NUMBER_TEST_NAME)
+        await user.should_see(DIRECTION_DESCRIPTION_TEST_NAME)
+        await user.should_see(HAS_BICYCLE_LANE_TEST_NAME)
+        await user.should_see(IS_BICYCLE_COUNTING_TEST_NAME)
+        await user.should_see(REMARK_TEST_NAME)
+        await user.should_see("Geeokoordinate")
+        await user.should_see(COORDINATE_X_TEST_NAME)
+        await user.should_see(COORDINATE_Y_TEST_NAME)
+
+    @pytest.mark.asyncio
+    async def test_input_field_updates_viewmodel(
+        self, user: User, svz_metadata_form: SVZMetadataForm, viewmodel: Mock
+    ) -> None:
+        """Test that entering text in an input field updates the viewmodel."""
+
+        @ui.page(ENDPOINT_NAME)
+        def page() -> None:
+            svz_metadata_form.build()
+
+        await user.open(ENDPOINT_NAME)
+
+        # Enter text in the TK Number field
+        user.find(TK_NUMBER_TEST_NAME).type(TK_NUMBER_VALUE)
```

Comment:
After all form fields have markers set, the markers should be used to find the correct input elements.